Mono.Cecil.Cil/Symbols.cs

```diff
@@ -802,6 +809,45 @@ public ISymbolReader GetSymbolReader (ModuleDefinition module, string fileName)
 			return null;
 		}
 
+		public static SymbolReaderKind GetSymbolReaderKind(string fileName)
+		{
+			try
+			{
+				using (var assemblyDefinition = AssemblyDefinition.ReadAssembly(fileName))
+				{
+					if (assemblyDefinition.Modules.Count <= 0)
+						return SymbolReaderKind.None;
+
+					var module = assemblyDefinition.Modules[0];
+					var provider = new DefaultSymbolReaderProvider();
+					return provider.GetSymbolReaderKind(module, fileName);
+				}
+			}
+			catch (Exception)
+			{
+				return SymbolReaderKind.None;
+			}
+		}
+
+		public SymbolReaderKind GetSymbolReaderKind (ModuleDefinition module, string fileName)
+		{
+			if (module.Image.HasDebugTables ())
+				return SymbolReaderKind.None;
+
+			var pdb_file_name = Mixin.GetPdbFileName (fileName);
+
+			if (File.Exists (pdb_file_name))
+				return Mixin.IsPortablePdb (Mixin.GetPdbFileName (fileName))
+					? SymbolReaderKind.PortablePdb
+					: SymbolReaderKind.NativePdb;
+
+			var mdb_file_name = Mixin.GetMdbFileName (fileName);
+			if (File.Exists (mdb_file_name))
+				return SymbolReaderKind.Mdb;
+
+			return SymbolReaderKind.None;
```

Comment:
I think this function is wrong because it doesn't try to read the debug info files. That can fail is they don't match the module.